server/endpoint/health/searcher/endpoint_test.go

```diff
@@ -0,0 +1,149 @@
+			expectedHealth: "green",
+		},
+	}
+
+	for i, tc := range testCases {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			var err error
+
+			// Mock k8s api server handling `kubectl get aws/azure/kvmconfig <clusterID>`.
+			k8sAPIMockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Set("Content-Type", "application/json; charset=utf-8")
+				w.WriteHeader(http.StatusOK)
+				w.Write([]byte(tc.k8sAPIResponse))
+			}))
+			defer k8sAPIMockServer.Close()
+
+			var restConfig *rest.Config
+			{
+				c := k8srestconfig.Config{
+					Logger: microloggertest.New(),
+
+					Address:   k8sAPIMockServer.URL,
+					InCluster: false,
+				}
+
+				restConfig, err = k8srestconfig.New(c)
+				if err != nil {
+					t.Fatal("expected", nil, "got", err)
+				}
+			}
+
+			k8sClient, err := kubernetes.NewForConfig(restConfig)
+			if err != nil {
+				t.Fatal("expected", nil, "got", err)
+			}
+
+			g8sClient, err := versioned.NewForConfig(restConfig)
+			if err != nil {
+				t.Fatal("expected", nil, "got", err)
+			}
+
+			f := flag.New()
+			v := viper.New()
+			v.Set(f.Service.Provider.Kind, "kvm")
+
+			// Create health service.
+			var healthService *health.Service
+			{
+				healthConfig := health.Config{
+					Flag:      f,
+					Viper:     v,
+					K8sClient: k8sClient,
+					G8sClient: g8sClient,
+					Logger:    microloggertest.New(),
+				}
+
+				healthService, err = health.New(healthConfig)
+				if err != nil {
+					t.Fatal("Error creating health service: ", err)
+				}
+			}
+
+			// Create health endpoint.
+			var endpoint *Endpoint
+			{
+				c := Config{
+					Logger:     microloggertest.New(),
+					Middleware: &middleware.Middleware{},
+					Service: &service.Service{
+						Health: healthService,
+					},
+				}
+				endpoint, err = New(c)
+				if err != nil {
+					t.Fatal("Error on endpoint creation: ", err)
+				}
+			}
+
+			req, err := http.NewRequest(Method, Path, nil)
+			if err != nil {
+				t.Fatal("Unexpected HTTP error:", err)
+			}
+
+			ctx := req.Context()
+			type contextKey string
+			ctx = context.WithValue(ctx, contextKey("Content-Type"), "application/json; charset=utf-8")
```

Comment:
Is it required to create the context like that? Can't it be simply `context.Background()`? (I didn't check the code 🙈 )

Reply:
I think you're right. And this endpoint doesn't even use the request, only pulls the id from the path using mux. Fixed!